openapi3/refs.go

```diff
@@ -559,6 +560,10 @@ func (x *ResponseRef) JSONLookup(token string) (interface{}, error) {
 // SchemaRef represents either a Schema or a $ref to a Schema.
 // When serializing and both fields are set, Ref is preferred over Value.
 type SchemaRef struct {
+	// Extensions only captures fields starting with 'x-' as no other fields
+	// are allowed by the openapi spec.
+	Extensions map[string]interface{}
```

Comment:
I'm not sure this should live here: this field already exists in the `.Value`. Or maybe it'd make more sense to move `.Value.Extensions` here, alongside `.Ref` `.Value` and `.extra`. This makes sense to me. You?

Reply:
(this change alone would be its own PR)

Reply:
I did a quick test and it looks like `ref.Value` objects are reused for all the places they are referenced. Thus updating them will change them everywhere.

I personally think we should keep it as proposed here. This allows users to know where the extension came from. They can use `schemaRef.JSONLookup("x-....")` to do the most common thing, check ref, then fallback to Value.